src/app/data-planes/data/DataplaneNetworking.ts

```diff
@@ -0,0 +1,103 @@
+import type {
+  DataplaneGateway as PartialDataplaneGateway,
+  DataplaneInbound as PartialDataplaneInbound,
+  DataplaneNetworking as PartialDataplaneNetworking,
+  DataplaneOutbound as PartialDataplaneOutbound,
+} from '@/types/index.d'
+
+type Connection = {
+  name: string
+  service: string
+  protocol: string
+}
+export type DataplaneInbound = PartialDataplaneInbound & Connection & {
+  address: string
+  state: NonNullable<PartialDataplaneInbound['state']>
+  addressPort: string
+  serviceAddressPort: string
+  listenerAddress: string
+}
+export type DataplaneOutbound = PartialDataplaneOutbound & Connection & {
+}
+
+export type DataplaneGateway = PartialDataplaneGateway & {}
+export type DataplaneNetworking = Omit<PartialDataplaneNetworking, 'inbound' | 'outbound'> & {
+  inboundAddress: string
+  inbounds: DataplaneInbound[]
+  outbounds: DataplaneOutbound[]
+  type: 'sidecar' | 'gateway'
+}
+const DataplaneOutbound = {
+  fromObject(item: PartialDataplaneOutbound): DataplaneOutbound {
+    return {
+      ...item,
+      name: item.tags['kuma.io/service'],
+      service: item.tags['kuma.io/service'],
+      protocol: item.tags['kuma.io/protocol'] ?? 'tcp',
+    }
+  },
+  fromCollection(items: PartialDataplaneOutbound[]): DataplaneOutbound[] {
+    return Array.isArray(items) ? items.map(item => DataplaneOutbound.fromObject(item)) : []
+  },
+}
+export const DataplaneNetworking = {
+  fromObject(networking: PartialDataplaneNetworking): DataplaneNetworking {
+    // remove singular inbound/outbound to be replaced with plural versions
+    const { inbound, outbound, ...rest } = networking
+
+    const inbounds = Array.isArray(inbound) ? inbound : []
+
+    // outbounds are only present here on a universal DDP without transparent
+    // proxying
+    const outbounds = Array.isArray(outbound) ? outbound : []
+    const type = typeof networking.gateway === 'undefined' || networking.gateway?.type !== 'BUILTIN' ? 'sidecar' : 'gateway'
+    return {
+      ...rest,
+      type,
+      // used for a lookup for inbounds on the result of the envoy /stats endpoint
+      inboundAddress: type === 'gateway' ? networking.address : 'localhost',
+      //
+      // if we are a builtin gateway fill in as much as we can for a single inbound
+      // we can 'clone' this later if we find out individual information for each inbound
+      // i.e. this acts as a template
+      inbounds: type === 'gateway' && typeof networking.gateway !== 'undefined'
+        ? [{
+          address: networking.address,
+          tags: networking.gateway.tags,
+          name: networking.gateway.tags['kuma.io/service'],
+          service: networking.gateway.tags['kuma.io/service'],
+          protocol: networking.gateway.tags['kuma.io/protocol'] ?? 'tcp',
+          state: 'Ready',
```

Comment:
What about creating a constant `Ready` or enum `state.ready`  for future maintenance?

Reply:
This is one of those things where GH reviews can't show you things! If you check it out and play with it you shouldn't be able to set this `state:` property to anything other than one of the 3 strings defined in our single source of truth (the typescript types) `Ready`, `NotReady`, `Ignored`? So the type definition is essentially our enum (which will eventually be auto generated)

It should all be tied together via the types to prevent typos, even the places where we check for equality.